interop-java/src/main/java/scalaz/zio/java/data/Either.java

```diff
@@ -0,0 +1,65 @@
+package scalaz.zio.java.data;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+public abstract class Either<L, R> {
+
+    public static class Left<L, R> extends Either<L, R> {
+        final L value;
```

Comment:
private

Reply:
I actually think this can be public. It doesn't seem harmful for me to let people access it if they want. Maybe they want to do something we haven't really thought of.